scipy/optimize/tests/test_linprog.py

```diff
@@ -687,6 +687,75 @@ def cb(xk, **kwargs):
         assert_(callback_complete[0])
         assert_allclose(last_xk[0], res.x)
 
+    def test_bug_8662(self):
+        # scipy.linprog returns incorrect optimal result for constraints using
+        # default bounds, but, correct if boundary condition as constraint.
+        # https://github.com/scipy/scipy/issues/8662
+        c = [-10, 10, 6, 3]
+        A = [
+            [8, -8, -4, 6],
+            [-8, 8, 4, -6],
+            [-4, 4, 8, -4],
+            [3, -3, -3, -10]
+        ]
+        b = [9, -9, -9, -4]
+        bounds = [(0, None), (0, None), (0, None), (0, None)]
+
+        res1 = linprog(c, A, b, bounds=bounds)
+        _assert_success(res1, 36.0000000000)
+
+        # Set boundary condition as a constraint
+        A.append([0, 0, -1, 0])
+        b.append(0)
+        bounds[2] = (None, None)
+
+        res2 = linprog(c, A, b, bounds=bounds)
+        _assert_success(res2, 36.0000000000)
+        assert_allclose(res1.x, res2.x)
+
+    def test_bug_6690_with_simplex(self):
+        # SciPy violates bound constraint despite result status being success
+        # when the simplex method is used.
+        # https://github.com/scipy/scipy/issues/6690
+
+        A_eq = np.array([[0, 0, 0, 0.93, 0, 0.65, 0, 0, 0.83, 0]])
+        b_eq = np.array([0.9626])
+        A_ub = np.array([
+            [0, 0, 0, 1.18, 0, 0, 0, -0.2, 0, -0.22],
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0.43, 0, 0, 0, 0, 0, 0],
+            [0, -1.22, -0.25, 0, 0, 0, -2.06, 0, 0, 1.37],
+            [0, 0, 0, 0, 0, 0, 0, -0.25, 0, 0]
+        ])
+        b_ub = np.array([0.615, 0, 0.172, -0.869, -0.022])
+        bounds = np.array([
+            [-0.84, -0.97, 0.34, 0.4, -0.33, -0.74, 0.47, 0.09, -1.45, -0.73],
+            [0.37, 0.02, 2.86, 0.86, 1.18, 0.5, 1.76, 0.17, 0.32, -0.15]
+        ]).T
+        c = np.array(
+            [-1.64, 0.7, 1.8, -1.06, -1.16,0.26, 2.13, 1.53, 0.66, 0.28]
+        )
+
+        res = linprog(
+            c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
+            bounds=bounds, method='simplex'
+        )
+        _assert_success(res, -1.19099999999)
+        assert np.less_equal(bounds[:, 0], res.x).all()
+        assert np.greater_equal(bounds[:, 1] + 1e-12, res.x).all()
+
+    def test_bug_7044_simplex(self):
```

Comment:
Same here. Rather than adding this test here, `test_magic_square_bug_7044` can be moved out of the `BaseTestLinprogIP` class and into the `LinprogCommonTests` class.